action/protocol/rewarding/protocol.go

```diff
@@ -215,16 +215,11 @@ func (p *Protocol) Handle(
 		return p.settleUserAction(ctx, sm, uint64(iotextypes.ReceiptStatus_Success), si, nil, rlog)
 	case *action.ClaimFromRewardingFund:
 		si := sm.Snapshot()
-		claimTo := act.Address()
-		if claimTo == "" {
-			claimTo = protocol.MustGetActionCtx(ctx).Caller.String()
+		addr := act.ClaimAccount()
+		if addr == nil {
+			addr = protocol.MustGetActionCtx(ctx).Caller
 		}
-		claimToAddr, err := address.FromString(claimTo)
-		if err != nil {
-			log.L().Debug("Invalid claim to address", zap.Error(err))
-			return p.settleUserAction(ctx, sm, uint64(iotextypes.ReceiptStatus_Failure), si, nil)
-		}
```

Comment:
this address check can (and should) be done before `Handle()` in the action's `LoadProto()` and `SanityCheck()`, this is consistent with other actions like candidate register